### PR TITLE
1.7.0

### PR DESCRIPTION
**Pull request by goranalkovic-infinum**
Changes:
- Updated dependencies.
- `Repeater`, `Draggable`, `DraggableList` and `ItemCollection` will now behave better when `undefined` is passed as value.
- Fixed overlap issues with `Select` components.
- Improved `ColorPicker` grouping logic and made grouped menu more compact.
- Added `MiniResponsive` for cases you need to use a `Responsive` in a limited space. Works best with `OptionSelect` or other `Button`-like components!
- Fixed `Spacer` not applying `className` in some cases.
- Tweaked `MediaPlaceholder` colors a bit.
- `MediaPlaceholder` can now have any child items passed. They are rendered below the label and icon.

## `MiniResponsive`
with label:
image

or standalone:
image

expanded state:
image

### CODE REVIEW

**Review by iobrado — CHANGES_REQUESTED** [review]
Found some commented out code, but other than that, looks good to me 🙂

File: lib/components/select/styles.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const multiValueRemoveStyles =
 const clearIndicatorStyles = 'es-uic-text-gray-500 es-uic-p-1 es-uic-rounded-md hover:bg-red-50 hover:text-red-800 es-uic-transition';
 const dropdownIndicatorStyles =
 	'es-uic-text-gray-500 es-uic-px-1 group-hover:es-uic-text-black [&>svg]:es-uic-transition-transform [&>svg]:es-uic-duration-500 [&>svg]:es-uic-size-5.5';
-const menuStyles = 'es-uic-z-50 es-uic-rounded-md es-uic-border es-uic-border-gray-200 es-uic-bg-white es-uic-shadow-lg es-uic-mt-1 es-uic-text-sm es-uic-overflow-x-hidden';
+const menuStyles = 'es-uic-rounded-md es-uic-border es-uic-border-gray-200 es-uic-bg-white es-uic-shadow-lg es-uic-mt-1 es-uic-text-sm es-uic-overflow-x-hidden';
```
Comment (piqusy) [inline lib/components/select/styles.js:22]: 👀 Going back to the faulty version?

File: lib/components/draggable-list/draggable-list.jsx [hunk 2]
```diff
@@ -81,7 +81,11 @@ export const DraggableList = (props) => {
 		...rest
 	} = props;
 
-	const items = fixIds(rawItems, itemIdBase);
+	if (typeof rawItems === 'undefined' || rawItems === null || !Array.isArray(rawItems)) {
+		console.warn(__("DraggableList: 'items' are not an array or are undefined!", 'eightshift-ui-components'));
```
Comment (kancijan) [inline lib/components/draggable-list/draggable-list.jsx:85]:
```suggestion
		console.warn(__("DraggableList: The 'items' property is either not an array or is undefined!", 'eightshift-ui-components'));
```

File: lib/components/responsive/mini-responsive.jsx [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+		noModeSelect,
+
+		children,
+
+		hidden,
+
+		innerContentAlign = 'start',
+	} = props;
+
+	const breakpoints = rawBreakpoints.slice(1);
```
Comment (piqusy) [inline lib/components/responsive/mini-responsive.jsx:100]:
```suggestion
	const breakpoints = rawBreakpoints?.slice(1);
```

File: lib/components/responsive/mini-responsive.jsx [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+				text={
+					<div className='es-uic-max-w-64 es-uic-p-1'>
+						<span className='es-uic-block es-uic-text-balance es-uic-font-semibold es-uic-tabular-nums'>{__('Default', 'eightshift-ui-components')}</span>
+
+						<span className='es-uic-block es-uic-text-balance es-uic-tabular-nums'>
+							{!firstMobileFirstOverride && !lastDesktopFirstOverride && __('Always applied, regardless of browser width.', 'eightshift-ui-components')}
+
+							{firstMobileFirstOverride &&
+								!isDesktopFirst &&
+								sprintf(__('Applies when the browser width is %dpx or narrower.', 'eightshift-ui-components'), breakpointData[firstMobileFirstOverride] - 1)}
```
Comment (piqusy) [inline lib/components/responsive/mini-responsive.jsx:135]:
```suggestion
								sprintf(__('Applies when the browser width is %dpx or less.', 'eightshift-ui-components'), breakpointData[firstMobileFirstOverride] - 1)}
```

**Review by piqusy — CHANGES_REQUESTED** [review]
Nice work 👏🏼 

Left a few suggestions

File: lib/components/select/styles.js (hunk 1 above)
Comment (goranalkovic-infinum) [reply lib/components/select/styles.js:22]: Z-index never did anything, so no point in having it 😄 
The actual solution is in the commit 😉

**Review by piqusy — APPROVED** [review]

File: lib/components/draggable-list/draggable-list.jsx (hunk 2 above)
Comment (kancijan) [reply lib/components/draggable-list/draggable-list.jsx:85]: Apply to other places, too.

**Review by kancijan — APPROVED** [review]
Left a suggestion

**Review by iobrado — APPROVED** [review]

**Pull request merged**